### ISSUE TITLE
Add support for IToggleProvider

### ISSUE THREAD

**Issue by chiaramooney** [microsoft/react-native-windows#13329]
## Documentation
IToggleProvider documentation: https://learn.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-itoggleprovider

## Behavior
Components of control type CheckBox must implement the IToggleProvider. Components of control type Button, ListItem, MenuItem, and TreeItem should conditionally implement the IToggleProvider (See Documentation: https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-controlpatternmapping). The toggle state of the control should be used as input for the IToggleProvider data. 

## Implementation Plan
We should:

Implement the IToggleProvider APIs in CompositionDynamicAutomationProvider.
Set the list of controls who implement the IToggleProvider to be Button, ListItem, MenuItem, and TreeItem under certain conditions. To get more information on the conditions visit the page for the individual UIA control types.

For additional reference, see implementation of another provider on Fabric: https://github.com/microsoft/react-native-windows/pull/11874

## To Be Clarified
